codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -832,8 +846,9 @@ public void flowController(F flowController) {
         }
 
         private void checkNewStreamAllowed(int streamId) throws Http2Exception {
-            if (goAwaySent() || goAwayReceived()) {
-                throw connectionError(PROTOCOL_ERROR, "Cannot create a stream since the connection is going away");
+            if (goAwayReceived()) {
```

Comment:
@Scottmitch yes, I'm fine with leaving this as-is for now.  I'm not sure yet whether we need to make this extensible at the low level or just leave the heavy lifting to the application.  Let's let the dust settle and raise the issue if needed.  Sound good?